src/parser/postgresparser.cpp

```diff
@@ -288,36 +288,23 @@ parser::GroupByDescription* PostgresParser::GroupByTransform(List* group,
   result->columns = new std::vector<expression::AbstractExpression*>();
   for (auto cell = group->head; cell != nullptr; cell = cell->next) {
     Node* temp = reinterpret_cast<Node*>(cell->data.ptr_value);
-    switch (temp->type) {
-      case T_ColumnRef:
```

Comment:
Here, group by columns should only be either a tuple value expression or complex expressions like a + b. In your new code, other expressions will also be parsed and passed to the optimizer. Then, it becomes the optimizer's responsibility to throw the errors. But originally, the errors will come out of parser. Same problems for other places.